collect_app/src/main/java/org/odk/collect/android/geo/MapFragment.java

```diff
@@ -206,4 +209,11 @@ void addTo(
     interface FeatureListener {
         void onFeature(int featureId);
     }
+
+    default File getReferenceLayerFile(Bundle config) {
```

Comment:
It feels to me like this could be static util helper method (that probably takes a `StoragePathProvider` and `Bundle` as args). I'd generally think about a default implementation adding something to the external interface where as this seems more like a helper for the `MapFragment` implementations. What do you think?

Either way I think this piece of logic should have a test to drive out that it checks for file existance as well as grabbing the layer path.

Reply:
Ok I moved getReferenceLayerFile() to GeoUtils. When it comes to tests... getAbsoluteOfflineMapLayerPath() has its own tests so does that make sense to write a test just to check the return statement (file existence):
```
return path != null && new File(path).exists()
                ? new File(path)
                : null;
```
?

Reply:
Yeah it's a small decision but it's an important one so good to have a test to make sure it stays in place.

Reply:
Ok done, I added tests.